virtualizarr/xarray.py

```diff
@@ -50,37 +51,88 @@ def open_virtual_dataset(
         Currently can only be ManifestArray, but once VirtualZarrArray is implemented the default should be changed to that.
     """
 
-    # this is the only place we actually always need to use kerchunk directly
-    vds_refs = kerchunk.read_kerchunk_references_from_file(
-        filepath=filepath,
-        filetype=filetype,
-    )
+    if drop_variables is None:
+        drop_variables = []
+
+    if virtual_array_class is not ManifestArray:
+        raise NotImplementedError()
+
+    if filetype == "zarr_v3":
+        # TODO is there a neat way of auto-detecting this?
+        return open_virtual_dataset_from_v3_store(storepath=filepath, drop_variables=drop_variables, indexes=indexes)
+    else:
+        # this is the only place we actually always need to use kerchunk directly
+        vds_refs = kerchunk.read_kerchunk_references_from_file(
+            filepath=filepath,
+            filetype=filetype,
+        )
+
+        if indexes is None:
+            # add default indexes by reading data from file
+            # TODO we are reading a bunch of stuff we know we won't need here, e.g. all of the data variables...
+            # TODO it would also be nice if we could somehow consolidate this with the reading of the kerchunk references
+            ds = xr.open_dataset(filepath)
+            indexes = ds.xindexes
+            ds.close()
+
+        vds = dataset_from_kerchunk_refs(
+            vds_refs,
+            drop_variables=drop_variables,
+            virtual_array_class=virtual_array_class,
+            indexes=indexes,
+        )
+
+        # TODO we should probably also use vds.set_close() to tell xarray how to close the file we opened
+
+        return vds
+
+
+def open_virtual_dataset_from_v3_store(
+    storepath: str,
+    drop_variables: List[str],
+    indexes: Optional[Mapping[str, Index]],
+) -> xr.Dataset:
+    """
+    Read a Zarr v3 store and return an xarray Dataset containing virtualized arrays.
+    """
+    _storepath = Path(storepath)
+
+    ds_attrs = attrs_from_zarr_group_json(_storepath / "zarr.json")
+
+    # TODO recursive glob to create a datatree
+    vars = {}
+    for array_dir in _storepath.glob("*/"):
```

Comment:
It seems that when run locally (on MacOS), a `pathlib.Path.glob("*/")` call only returns  directories (as the pathlib docs say it will), but for some reason when run in this CI the glob will include files too. I've hacked around this by excluding any paths for which `.is_file()` is True.